internal/client/aws.go

```diff
@@ -0,0 +1,231 @@
+)
+
+func NewAwsCli(ctx context.Context, cfg *config.Config) (*AwsCli, error) {
+	cliCfg, err := cfg.GetAWSConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get AWS cli context: %w", err)
+	}
+
+	client := ec2.NewFromConfig(cliCfg)
+	awsCli := &AwsCli{
+		cfg:    cfg,
+		client: *client,
+		region: cfg.Region,
+	}
+
+	return awsCli, nil
+}
+
+type AwsCli struct {
+	cfg *config.Config
+
+	client ec2.Client
+	region string
+}
+
+func (a *AwsCli) StartInstance(ctx context.Context, vmName string) error {
+	_, err := a.client.StartInstances(ctx, &ec2.StartInstancesInput{
+		InstanceIds: []string{vmName},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to start instance: %w", err)
+	}
+
+	return nil
+}
+
+func (a *AwsCli) StopInstance(ctx context.Context, vmName string) error {
+	_, err := a.client.StopInstances(ctx, &ec2.StopInstancesInput{
+		InstanceIds: []string{vmName},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to stop instance: %w", err)
+	}
+
+	return nil
+}
+
+func (a *AwsCli) FindInstances(ctx context.Context, controllerID, instanceName string) (*[]types.Instance, error) {
+	resp, err := a.client.DescribeInstances(ctx, &ec2.DescribeInstancesInput{
+		Filters: []types.Filter{
+			{
+				Name:   aws.String("tag:GARM_CONTROLLER_ID"),
+				Values: []string{controllerID},
+			},
+			{
+				Name:   aws.String("tag:Name"),
+				Values: []string{instanceName},
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to find instances by tags: %w", err)
+	}
+
+	var instances []types.Instance
+	for _, reserv := range resp.Reservations {
+		instances = append(instances, reserv.Instances...)
+	}
+
+	return &instances, nil
+}
+
+func (a *AwsCli) FindOneInstance(ctx context.Context, controllerID, instanceName string) (*types.Instance, error) {
+	resp, err := a.FindInstances(ctx, controllerID, instanceName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find instance %s: %w", instanceName, errors.ErrNotFound)
+	}
+
+	if len(*resp) > 1 {
+		return nil, fmt.Errorf("found more than one instance with name %s", instanceName)
+	}
+
+	return &(*resp)[0], nil
+
+}
+
+// Describes the specified instances or all instances. If you specify instance
+// IDs, the output includes information for only the specified instances. If you
+// specify filters, the output includes information for only those instances that
+// meet the filter criteria.
+func (a *AwsCli) GetInstance(ctx context.Context, instance string) (*types.Instance, error) {
+	resp, err := a.client.DescribeInstances(ctx, &ec2.DescribeInstancesInput{
+		InstanceIds: []string{instance},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to get instance: %w", err)
+	}
+
+	var instances []types.Instance
+	for _, reserv := range resp.Reservations {
+		instances = append(instances, reserv.Instances...)
+	}
+
+	if len(instances) == 0 {
+		return nil, fmt.Errorf("no such instance %s: %w", instance, errors.ErrNotFound)
+	}
+
+	return &instances[0], nil
```

Comment:
no need to return a pointer here. Just return the instance of the `types.Instance{}` type.